### PR TITLE
build GTK icon caches for some packages

### PR DESCRIPTION
**Pull request by romildo**
###### Motivation for this change

- maia-icon-theme: 2016-09-16 -> 2018-02-24
- paper-icon-theme: build GTK icon cache for Paper-Mono-Dark
- efl: build GTK icon cache
- gnome-themes-extra: build GTK icon cache
- mate-icon-theme-faenza: build GTK icon cache
- mate-themes: build GTK icon cache

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by romildo** [comment]
@GrahamcOfBorg build maia-icon-theme paper-icon-theme enlightenment.efl gnome3.gnome-themes-extra mate.mate-icon-theme-faenza mate.mate-themes

**Comment by jtojnar** [comment]
Should we do this? Icon cache is supposed to be global state, see the discussion in https://github.com/NixOS/nixpkgs/pull/45058

**Comment by romildo** [comment]
>Should we do this? Icon cache is supposed to be global state

@jtojnar I am not sure. But I am getting warnings with `xfce4-appearance-settings` with the icon themes that do not have a cache:

screenshot_2018-10-08_11-23-55

**Comment by romildo** [comment]
Currently for most icon theme packages the icon cache is created when the package is built. The exception is the `hicolor` icon theme, whose cache is built by `nixpkgs/nixos/modules/config/system-path.nix`. Some icon themes though  are missing a cache. The goal of this PR is to provide the cache for some of them.

As far as I know no other infra-structure is currently provided to build the icon caches.

**Comment by romildo** [comment]
This solution is commented on https://github.com/NixOS/nixpkgs/pull/27458, for instance.

**Comment by michaelpj** [comment]
As I think we concluded in the other PR, we actually want to build the icon cache both per-theme and globally, so this is also desirable.

**Comment by michaelpj** [comment]
I wonder if we could package this step up into a hook that could be added to `buildInputs`? That's probably a separate refactoring, though.

File: pkgs/desktops/mate/mate-themes/default.nix [hunk 1]
```diff
@@ -16,6 +16,10 @@ stdenv.mkDerivation rec {
 
   propagatedUserEnvPkgs = [ gtk-engine-murrine ];
 
+  postInstall = ''
+    "${gtk3.out}/bin/gtk-update-icon-cache" "$out"/share/icons/ContrastHigh
```
Comment (matthewbauer) [inline pkgs/desktops/mate/mate-themes/default.nix]: To support cross compilation you should put gtk3 in nativeBuildInputs and just use gtk-update-icon-cache

**Review by matthewbauer — CHANGES_REQUESTED** [review]
Otherwise looks good!

File: pkgs/desktops/mate/mate-themes/default.nix (hunk 1 above)
Comment (romildo) [reply pkgs/desktops/mate/mate-themes/default.nix]: Done.

**Pull request merged**